federation/cluster/common.sh

```diff
@@ -29,13 +29,20 @@ HOST_CLUSTER_ZONE="${FEDERATION_HOST_CLUSTER_ZONE:-}"
 # If $HOST_CLUSTER_ZONE isn't specified, arbitrarily choose
 # last zone as the host cluster zone.
 if [[ -z "${HOST_CLUSTER_ZONE}" ]]; then
-  E2E_ZONES_ARR=($E2E_ZONES)
-  HOST_CLUSTER_ZONE=${E2E_ZONES_ARR[-1]}
+  E2E_ZONES_ARR=(${E2E_ZONES:-})
+  # The test condition is a bash-ism for non-empty array check.
+  if [[ ${E2E_ZONES_ARR[@]:+${E2E_ZONES_ARR[@]}} ]]; then
```

Comment:
That doesn't work because we set -u. When -u is set, empty array is treated as an unbound variable.